frontend/src/react-query/api/transform.tsx

```diff
@@ -0,0 +1,115 @@
+  type GetTransformRequest as GetTransformRequestDataPlane,
+  GetTransformRequestSchema as GetTransformRequestSchemaDataPlane,
+  type GetTransformResponse,
+  type ListTransformsRequest as ListTransformsRequestDataPlane,
+  ListTransformsRequestSchema as ListTransformsRequestSchemaDataPlane,
+  type ListTransformsResponse,
+} from 'protogen/redpanda/api/dataplane/v1/transform_pb';
+import { MAX_PAGE_SIZE, type MessageInit, type QueryOptions } from 'react-query/react-query.utils';
+import { useInfiniteQueryWithAllPages } from 'react-query/use-infinite-query-with-all-pages';
+import { TOASTS, formatToastErrorMessageGRPC, showToast } from 'utils/toast.utils';
+
+export const useListTransformsQuery = (
+  input?: MessageInit<ListTransformsRequestDataPlane>,
+  options?: QueryOptions<GenMessage<ListTransformsRequest>, ListTransformsResponse>,
+) => {
+  const listTransformsRequestDataPlane = create(ListTransformsRequestSchemaDataPlane, {
+    pageSize: MAX_PAGE_SIZE,
+    pageToken: '',
+    ...input,
+  });
+
+  const listTransformsRequest = create(ListTransformsRequestSchema, {
+    request: listTransformsRequestDataPlane,
+  }) as MessageInit<ListTransformsRequest> & Required<Pick<MessageInit<ListTransformsRequest>, 'request'>>;
+
+  const listTransformsResult = useInfiniteQueryWithAllPages(listTransforms, listTransformsRequest, {
+    pageParamKey: 'request',
+    enabled: options?.enabled,
+    getNextPageParam: (lastPage) =>
+      lastPage?.response?.nextPageToken
+        ? {
+            ...listTransformsRequestDataPlane,
+            pageToken: lastPage.response?.nextPageToken,
+          }
+        : undefined,
+  });
+
+  const transforms = listTransformsResult?.data?.pages?.flatMap(({ response }) => response?.transforms);
+
+  return {
+    ...listTransformsResult,
+    data: {
+      transforms,
+    },
+  };
+};
+
+export const useGetTransformQuery = (
+  input?: MessageInit<GetTransformRequestDataPlane>,
+  options?: QueryOptions<GenMessage<GetTransformRequest>, GetTransformResponse>,
+) => {
+  const getTransformRequestDataPlane = create(GetTransformRequestSchemaDataPlane, {
+    name: input?.name,
+  });
+
+  const getTransformRequest = create(GetTransformRequestSchema, {
+    request: getTransformRequestDataPlane,
+  }) as MessageInit<GetTransformRequest> & Required<Pick<MessageInit<GetTransformRequest>, 'request'>>;
+
+  return useQuery(getTransform, getTransformRequest, {
+    enabled: options?.enabled !== false && input?.name !== '',
+  });
+};
+
+export const useDeleteTransformMutationWithToast = () => {
+  const queryClient = useQueryClient();
+
+  return useMutation(deleteTransform, {
+    onSuccess: async (_data, variables) => {
+      await queryClient.invalidateQueries({
+        queryKey: createConnectQueryKey({
+          schema: TransformService.method.listTransforms,
+          cardinality: 'infinite',
+        }),
+        exact: false,
+      });
+
+      showToast({
+        id: TOASTS.TRANSFORM.DELETE.SUCCESS,
```

Comment:
This ID is a big flaw in the cloud-ui I won't use it in here, it's better to generate a unique ID for instance.

Reply:
Fixed.